src/klipse/klipse_editors.cljs

```diff
@@ -139,16 +139,18 @@
         state (create-state :container container :result-element result-element :editor-args editor-args)]
     (when result-element (gdom/setTextContent result-element default-txt))
     (handle-events in-editor
-                   {:idle-msec idle-msec
-                    :on-should-eval #(eval-in-html-editor eval-fn result-element in-editor snippet-args state)})
+                   {:idle-msec      idle-msec
+                    :on-should-eval #(do
+                                       (on-edit-cb)
```

Comment:
why do you first call `on-edit-cb` and then eval the current snippet?
I would expect the evaluation of current snippet to occur first.

Reply:
It works because js is single threaded and pushing on a channel is similar to `setTimeout`.

Reply:
I think the code would me clearer if you switch the two forms. What do you think?